src/plugins/TimeTravelPlugin.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * @module plugins/TimeTravelPlugin
+ */
+import { observe } from '../utils/storeUtils';
+import { modifyLayer } from '../store/layers/layers.action';
+import { BaPlugin } from './BaPlugin';
+import { closeBottomSheet, openBottomSheet } from '../store/bottomSheet/bottomSheet.action';
+import { html } from 'lit-html';
+
+/**
+ * This plugin does the following time travel related things:
+ *
+ * - detect if the time travel slider should be visible
+ *
+ * - update the timestamp property of all suitable layers the timestamp property of the timeTravel s-o-s changes
+ *
+ *
+ * Note: This plugin must be registered **before** the `LayersPlugin`
+ * @class
+ * @author taulinger
+ */
+export class TimeTravelPlugin extends BaPlugin {
+	#currentGeoResourceId = null;
+
+	/**
+	 * @override
+	 */
+	async register(store) {
+		const findSuitableLayers = (layers) => {
+			return layers.filter((l) => l.visible).filter((l) => l.timestamp);
+		};
+
+		const onLayersChanged = (activeLayers) => {
+			/**
+			 * Check if we have one or more layers referencing the same timestamp
+			 * and check if they reference all the same GeoResource.
+			 * Only in that case we show the time travel component.
+			 */
+			const timestampSet = new Set(findSuitableLayers(activeLayers).map((l) => l.timestamp));
+			const geoResourceSet = new Set(findSuitableLayers(activeLayers).map((l) => l.geoResourceId));
+
+			if (timestampSet.size === 1 && geoResourceSet.size === 1) {
+				this.#currentGeoResourceId = [...geoResourceSet][0];
+				openBottomSheet(
+					html`<ba-time-travel-slider .timestamp=${[...timestampSet][0]} .geoResourceId=${this.#currentGeoResourceId}></ba-time-travel-slider>`
```

Comment:
if we relate this bottomSheet content to a specific id like'timetravel', there is no need to revisit this line later on (for instance, if the bottomSheet behavior evolves to free floating windows etc.)